internal/plugins/helm/v1/scaffolds/api.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Modifications copyright 2020 The Operator-SDK Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scaffolds
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/xenolf/lego/log"
+	"k8s.io/client-go/discovery"
+	crconfig "sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/kubebuilder/pkg/model"
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+	"sigs.k8s.io/kubebuilder/pkg/model/resource"
+	"sigs.k8s.io/kubebuilder/pkg/plugin/scaffold"
+
+	"github.com/operator-framework/operator-sdk/internal/kubebuilder/machinery"
+	"github.com/operator-framework/operator-sdk/internal/plugins/helm/v1/chartutil"
+	"github.com/operator-framework/operator-sdk/internal/plugins/helm/v1/scaffolds/templates"
+	"github.com/operator-framework/operator-sdk/internal/plugins/helm/v1/scaffolds/templates/crd"
+)
+
+var _ scaffold.Scaffolder = &apiScaffolder{}
+
+// apiScaffolder contains configuration for generating scaffolding for Go type
+// representing the API and controller that implements the behavior for the API.
+type apiScaffolder struct {
+	config *config.Config
+	opts   chartutil.CreateOptions
+}
+
+// NewAPIScaffolder returns a new Scaffolder for API/controller creation operations
+func NewAPIScaffolder(config *config.Config, opts chartutil.CreateOptions) scaffold.Scaffolder {
+	return &apiScaffolder{
+		config: config,
+		opts:   opts,
+	}
+}
+
+// Scaffold implements Scaffolder
+func (s *apiScaffolder) Scaffold() error {
+	switch {
+	case s.config.IsV3():
+		return s.scaffold()
+	default:
+		return fmt.Errorf("unknown project version %v", s.config.Version)
+	}
```

Comment:
Since this plugin already states via its `SupportedProjectVersions()` method that it only supports v3-alpha, is this switch necessary?

If I understand correctly, the kubebuilder plugin machinery will only allow this plugin to be used if it was invoked with one of the project versions that the plugin says it supports.

Reply:
good point 👍